lesson_2/employees.js

```diff
@@ -0,0 +1,41 @@
+const employees = [
+    {
+     firstName: 'Alex',
+     lastName: 'Smith',
+     age: 54,
+     salary: 10000,
+     position: 'Architect'
+    },
+    {
+     firstName: 'Gustav',
+     lastName: 'Andersen',
+     age: 31,
+     salary: 5000,
+     position: 'Software engineer'
+    },
+    {
+     firstName: 'Liz',
+     lastName: 'Taylor',
+     age: 20,
+     salary: 7000,
+     position: 'Manager'
+    }
+]
+
+const reduceCallback = function(acc, item) {
+    return acc + Number(item.salary)
+}
+
+const sortCallBack = function(a, b) {
+    return a.salary - b.salary
+}
+
+const filterCallBack = function(emp) {
+    return emp.salary > 4500 && emp.age > 25
+}
+
+const averageSalary = employees.reduce(reduceCallback, 0)/employees.length;
+const sortedBySalary = employees.sort(sortCallBack);
+const filterBySalaryAge = employees.filter(filterCallBack);
```

Comment:
Полагаю, что в этом случае также можно переместить _callback_ функции в соответствующие аргументы
```suggestion
const averageSalary = employees.reduce((acc, item) => acc + Number(item.salary), 0)/employees.length;
const sortedBySalary = employees.sort((a, b) => a.salary - b.salary);
const filterBySalaryAge = employees.filter(emp => emp.salary > 4500 && emp.age > 25);
```

Reply:
Здесь получается на длину строки не смотреть?

Reply:
В таком вопросе лучше положиться на "объективные" факторы: количество повторных использований кода в противовес к количеству однострочных функций наполняющих область видимости (и блокирующих [GC](https://learn.javascript.ru/garbage-collection))

Замечу, что мое **предложение** базируется на принципе [YAGNI](https://ru.m.wikipedia.org/wiki/YAGNI) и не противоречит другим разумным принципам 🙂